back/boxtribute_server/graph_ql/resolvers.py

```diff
@@ -398,13 +487,90 @@ def resolve_create_qr_code(*_, box_label_identifier=None):
+    authorize(permission="distro_event:write")
+    return change_distribution_event_state(distribution_event_id, new_state)
+
+
+@mutation.field("createDistributionEvent")
+@convert_kwargs_to_snake_case
+def resolve_create_distribution_event(*_, creation_input):
+    # TODO: Add permissions
+    authorize(permission="distro_event:write")
+    return create_distribution_event(user_id=g.user.id, **creation_input)
+
+
+@mutation.field("createDistributionSpot")
+@convert_kwargs_to_snake_case
+def resolve_create_distribution_spot(*_, creation_input=None):
+    authorize(permission="location:write")
+    return create_distribution_spot(
+        user_id=g.user.id, distribution_spot_input=creation_input
+    )
+
+
 @mutation.field("createBox")
 @convert_kwargs_to_snake_case
 def resolve_create_box(*_, creation_input):
     authorize(permission="stock:write")
     return create_box(user_id=g.user.id, **creation_input)
 
 
+@mutation.field("moveBoxToDistributionEvent")
+@convert_kwargs_to_snake_case
+def resolve_move_box_to_distribution_event(
+    mutation_obj, _, box_label_identifier, distribution_event_id
+):
+    authorize(permission="stock:write")
+    return move_box_to_distribution_event(box_label_identifier, distribution_event_id)
+
+
+@mutation.field("moveItemsFromBoxToDistributionEvent")
+@convert_kwargs_to_snake_case
+def resolve_move_items_from_box_to_distribution_event(
+    mutation_obj, _, box_label_identifier, distribution_event_id, number_of_items
+):
+    authorize(permission="unboxed_items_collection:write")
+    return move_items_from_box_to_distribution_event(
+        user_id=g.user.id,
+        box_label_identifier=box_label_identifier,
+        distribution_event_id=distribution_event_id,
+        number_of_items=number_of_items,
+    )
+
+
+@mutation.field("removePackingListEntryFromDistributionEvent")
+@convert_kwargs_to_snake_case
+def resolve_remove_packing_list_entry_from_distribution_event(
+    *_, packing_list_entry_id
+):
+
+    packing_list_entry = PackingListEntry.get(packing_list_entry_id)
+    if packing_list_entry is None:
+        # TODO: Discuss error handling approach
+        # * seems to be the first time we are doing GraphQLError throwings here
+        # * also discuss how to handle this in the frontend
+        # (e.g. should we return specific error codes that the FE then
+        # handles with messages?)
+        raise GraphQLError("Packing list entry not found")
+    distribution_event = (
+        DistributionEvent.select()
+        .where(DistributionEvent.id == packing_list_entry.distribution_event)
+        .get()
+    )
+    authorize(
+        permission="distro_event:write",
+        base_id=distribution_event.distribution_spot.base.id,
+    )
+    # Completed Events should not be mutable anymore
+    if distribution_event.state == DistributionEventState.Completed:
+        raise GraphQLError("Cannot move items to completed distribution event")
+    delete_packing_list_entry(packing_list_entry_id)
```

Comment:
For consistency with the other crud functions you could put the entire validation step into the `delete_...` function